modules/apps/headless/headless-builder/headless-builder-impl/src/main/java/com/liferay/headless/builder/internal/model/listener/APISchemaRelevantObjectEntryModelListener.java

```diff
@@ -107,12 +110,32 @@ private void _validate(ObjectEntry objectEntry) {
 		}
 	}
 
+	private void _validateCreate(ObjectEntry objectEntry) {
+		try {
+			
+			if (_objectEntryLocalService.fetchObjectEntry(
+					objectEntry.getExternalReferenceCode(),
+					objectEntry.getObjectDefinitionId()) != null) {
+
+				throw new ObjectEntryValuesException.InvalidObjectField(
+					null, "This external reference code is already in use",
+					"this-external-reference-code-is-already-in-use");
```

Comment:
do we already have a test that would cover this use case?

Reply:
not yet, working on it

Reply:
Done: https://github.com/liferay-headless/liferay-portal/pull/1905/commits/0bb2bf2b9f2275ad93120dc79a0fe13725dc3609